arrMethodsLab.js

```diff
@@ -0,0 +1,409 @@
+    return true
+  } else {
+    return false
+  }
+})
+
+let sameCheck2 = arr5.every((currentElem, index) => {
+  if (currentElem[index] === currentElem[index - 1]) {
+    return true
+  } else {
+    return false
+  }
+})
+
+let sameCheck3 = arr5.reduce((acc, currentElem) => {
+  if (acc > currentElem) {
+    return false
+  } else {
+    return true
+  }
+})
+
+let sameCheck = arr5.filter((currentElem) => {
+
+})
+
+console.log(sameCheck1)
+console.log(sameCheck2)
+console.log(sameCheck3)
+
+
+
+
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+//
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+//
+console.log(" ")
+
+// 5. Write a function that returns the sum of every element in an array.
+console.log("---------- #5 ----------")
+
+let arr6 = [1, 2, 3, 4, 5]
+
+let sum = 0
+arr6.forEach(num => sum += num)
+console.log(sum)
+
+// ```
+// Input: [1,2,3,4,5]
+// Output: 15
+// ```
+//
+console.log(" ")
+
+// 6. Write a function that finds largest number in an array.
+console.log("---------- #6 ----------")
+
+let arr7 = [1, 2, 6, 3, 4, 5]
+
+let largestInArr = arr7.reduce((acc, currentElem) => {
+  if (currentElem > acc) {
+    return currentElem
+  } else {
+    return acc
+  }
+})
+
+console.log(largestInArr)
+
+// ```
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+// ```
+//
+console.log(" ")
+
+// 7. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
+console.log("---------- #7 ----------")
+
+let arr8 = [1, 0, 0, 2, 0, 3, 0, 0, 0]
+
+let countZeroes = arr8.reduce((total, currentElem) => {
+  if (currentElem === 0) {
+    return total + 1
+  }
+}, 0)
+
+console.log(countZeroes)
+
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+  console.log(" ")
+//
+// 8. Write a function that doubles every element in an array.
+console.log("---------- #8 ----------")
+
+let arr9 = [1, 2, 3, 4]
+let doubled = []
+arr9.forEach((currentElem) => {
+  doubled.push(currentElem + currentElem)
+})
+
+console.log(doubled)
+
+// ```
+// Input: [1,2,3,4]
+// Output: [2,4,6,8]
+// ```
+
+console.log(" ")
+
+// 9. Write a function that finds the sum of every element in an array doubled
+console.log("---------- #9 ----------")
+
+let arr10 = [1, 2, 3, 4, 5]
+let sumOfAllDoubled = arr10.reduce((total, currentElem) => {
+  console.log(total, currentElem)
+  return total + (currentElem * 2)
+}, 0)
+
+console.log(sumOfAllDoubled)
+
+// ```
+// Input: [1,2,3,4]
+// Output: 20
+// ```
+//
+// 10. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.  Return a new array of every element of arr that can be evenly divided by divisor.
+console.log("---------- #10 ----------")
+
+
+let arr11 = [4,5,6,7,8,9,10,11,12,13,14,15]
+let divisor = 5
+
+const elementDivisibleBy = (divisor, arr) => {
+  return arr.filter((el) => {
+    if (el % 5 === 0) {
+      return el
+    }
+  })
+}
+
+console.log(elementDivisibleBy(divisor, arr11))
+
```

Comment:
So this code is actually hard coded .. What if the divisor I wanted to use is 2 or anything other than 5? How can we fix this?